src/components/StakingPage/StakePositionTable/index.tsx

```diff
@@ -0,0 +1,123 @@
+/** @jsx jsx */
+import { Box, Flex, jsx, Switch, Label } from 'theme-ui';
+import Table from 'rc-table';
+import { useCallback, useState, useMemo } from 'react';
+import styles from './styles';
+import type { ITableData, IStakePositionItem } from './types';
+import { createValueRenderer, expiryDateRenderer, vaultIconsRenderer } from './TableComponent';
+import { formatUnits } from 'ethers/lib/utils';
+const { Column } = Table;
+
+interface IProps {
+  data: IStakePositionItem[];
+  onRedeem: (index: number) => void;
+  onRestake: (index: number) => void;
+}
+
+export default function StakePositionTable(props: IProps) {
+  const { data, onRedeem, onRestake } = props;
+
+  const [showArchive, setShowArchive] = useState(false);
+
+  const handleArchive = useCallback(() => {
+    setShowArchive((state) => !state);
+  }, []);

```

Comment:
Don't we need to put setShowArchive in dependency?

Reply:
set hook callback function is working on the current state